src/web/components/HydrateOnce.tsx

```diff
@@ -32,5 +32,11 @@ export const HydrateOnce = ({ rootId, children, waitFor = [] }: Props) => {
 		end();
 	});
 	setAlreadyHydrated(true);
+	// @ts-ignore
+	if (window.STORYBOOK_ENV) {
+		window.addEventListener('storybook:reset', () =>
+			setAlreadyHydrated(false),
+		);
+	}
```

Comment:
Can this go in `useEffect`?

Reply:
I did it slightly differently in the end, removing the event listeners immediately before adding it again